src/entitysdk/util.py

```diff
@@ -135,3 +136,15 @@ def build_api_url(environment: DeploymentEnvironment) -> str:
         DeploymentEnvironment.staging: settings.staging_api_url,
         DeploymentEnvironment.production: settings.production_api_url,
     }[environment]
+
+
+def validate_filename_extension_consistency(path: Path, expected_extension: str) -> Path:
+    """Validate file path extension against expected extension."""
+    if path.suffix == expected_extension:
```

Comment:
We could probably ignore the upper/lower case differences.
We may also consider some additional sanitization/validation when importing data and when uploading in entitycore.
This is what we have now in the db (note for example asc/ASC):

```
extension|count|
---------+-----+
nwb      |10523|
swc      | 4626|
asc      | 4525|
h5       | 3276|
obj      | 1339|
nrrd     |  332|
json     |  238|
DAT      |  152|
abf      |  152|
mod      |  125|
hoc      |  118|
ASC      |   53|
xlsx     |    6|
```